pkgs/applications/audio/vcvrack/default.nix

```diff
@@ -0,0 +1,115 @@
+{ lib, stdenv, fetchFromGitHub, fetchgit, makeWrapper, gcc,
+  # buildInputs
+  curl, glew, glfw, gnome2, jansson, libsamplerate, libzip,
+  pkgconfig, rtaudio, rtmidi,
+  ... }:
+
+stdenv.mkDerivation rec {
+  name = "vcvrack-${version}";
+  version = "0.5.1";
+
+  githubUser = "VCVRack";
+
+  # fetch source
+  # ------------
+  # need to use fetchgit here to pull submodules as well
+  src = fetchgit {
+    url    = "https://github.com/${githubUser}/Rack.git";
+    rev    = "2c17b654afc7e5c6e8e70f41df9dd3f8ad9dde24";
+    sha256 = "0zy1splbir50iz12vlpxbpix1qx6lrjmqyb4fy7nz03sz8dij59a";
+  };
+
+  # fetch Plugins
+  # -------------
+  pluginFundamental = fetchgit {
```

Comment:
I tried that, but I couldn't make it work. For example I need the `pluginFundamental` variable later in the patch phase, I don't know how to dynamically create variables and such. If this would be Haskell I would know how to do it.